### PR TITLE
Don't create parent directory if it exists

### LINKED ISSUE #103: SQLiteDatabase.ensureFile logs an error, if the directory already exists
ensureFile should check that the directory doesn't exist before logging that there was an error creating the directory

### PR DESCRIPTION
**Pull request by EricAtPlanGrid**
This should fix #103.

### CODE REVIEW

File: sqlite-android/src/main/java/io/requery/android/database/sqlite/SQLiteDatabase.java [hunk 1]
```diff
@@ -826,8 +826,12 @@ private static void ensureFile(String path) {
         File file = new File(path);
         if (!file.exists()) {
             try {
-                if (!file.getParentFile().mkdirs()) {
-                    Log.e(TAG, "Couldn't mkdirs " + file);
+                if (!file.getParentFile().exists()) {
```
Comment (benasher44) [inline sqlite-android/src/main/java/io/requery/android/database/sqlite/SQLiteDatabase.java]: Maybe combine into one?
```suggestion
                if (!file.getParentFile().exists() && !file.getParentFile().mkdirs()) {
```

**Comment by benasher44** [comment]
I don't think it's worth testing log output, but it's probably good to make sure `ensureFile` is a codepath that's well-covered to avoid breakage.

File: sqlite-android/src/main/java/io/requery/android/database/sqlite/SQLiteDatabase.java [hunk 2]
```diff
@@ -826,8 +826,12 @@ private static void ensureFile(String path) {
         File file = new File(path);
         if (!file.exists()) {
             try {
-                if (!file.getParentFile().mkdirs()) {
-                    Log.e(TAG, "Couldn't mkdirs " + file);
+                if (!file.getParentFile().exists() && !file.getParentFile().mkdirs()) {
+                    // Fixes #103: Check parent directory's existence before
+                    // attempting to create.
+                    if (!file.getParentFile().mkdirs()) {
```
Comment (benasher44) [inline sqlite-android/src/main/java/io/requery/android/database/sqlite/SQLiteDatabase.java]:
```suggestion

```

File: sqlite-android/src/main/java/io/requery/android/database/sqlite/SQLiteDatabase.java [hunk 3]
```diff
@@ -826,8 +826,12 @@ private static void ensureFile(String path) {
...
+                        Log.e(TAG, "Couldn't mkdirs " + file);
+                    }
```
Comment (benasher44) [inline sqlite-android/src/main/java/io/requery/android/database/sqlite/SQLiteDatabase.java]:
```suggestion

```

File: sqlite-android/src/main/java/io/requery/android/database/sqlite/SQLiteDatabase.java [hunk 4]
```diff
@@ -826,8 +826,12 @@ private static void ensureFile(String path) {
         File file = new File(path);
         if (!file.exists()) {
             try {
-                if (!file.getParentFile().mkdirs()) {
-                    Log.e(TAG, "Couldn't mkdirs " + file);
+                if (!file.getParentFile().exists() && !file.getParentFile().mkdirs()) {
+                    // Fixes #103: Check parent directory's existence before
+                    // attempting to create.
+                    if (!file.getParentFile().mkdirs()) {
+                        Log.e(TAG, "Couldn't mkdirs " + file);
```
Comment (benasher44) [inline sqlite-android/src/main/java/io/requery/android/database/sqlite/SQLiteDatabase.java]:
```suggestion
                    Log.e(TAG, "Couldn't mkdirs " + file);
```

**Comment by npurushe** [comment]
Change generally looks good but the current state of the pr is calling `file.getParentFile().mkdirs()` twice will merge once that's resolved

**Pull request merged**